broker/amqp/queue.go

```diff
@@ -39,8 +41,16 @@ func newQueue(pipe *jobs.Pipeline, lsn func(event int, ctx interface{})) (*queue
 		return nil, fmt.Errorf("missing `queue` parameter on amqp pipeline")
 	}
 
+	isValidExchangeType := map[string]bool{"direct": true, "fanout": true, "topic": true, "headers": true}
+	exchangeType := pipe.String("exchange-type", "direct")
```

Comment:
I think it's better to have enum here for the exchange types. To be more obvious. For example in the `queue` struct:
`exchangeType    ExchangeType` where ExchangeType is:
```go
type ExchangeType string
```
```go
const (
	Direct  ExchangeType = "direct"
	Fanout  ExchangeType = "fanout"
	Topic   ExchangeType = "topic"
	Headers ExchangeType = "headers"
)
```
If needed, you can also implement `isValid` and `String` methods on this type.